components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -720,10 +703,14 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         }
 
         if (this.user) {
-            const updatedUser = await this.userDB.findUserById(this.user.id);
+            let updatedUser = await this.userDB.findUserById(this.user.id);
             if (updatedUser) {
                 this.user = updatedUser;
             }
+
+            if (this.userToTeamMigrationService.needsMigration(this.user)) {
+                this.user = await this.userToTeamMigrationService.migrateUser(this.user, true);
```

Comment:
I somehow feel this should be on a hot path like this, but it should not matter, really. Especially as checking the "already done" case is sync. :shrug: :+1:

Reply:
While testing, I got stuck on the "hang in there" screen, and had to reload two times to actually see the message. Maybe it blocked the `getLoggedInUser` call somehow? :thinking:

Reply:
> While testing, I got stuck on the "hang in there" screen, and had to reload two times to actually see the message. Maybe it blocked the getLoggedInUser call somehow? 🤔

Unlikely. Do you remember what you tested? I.e. was your user in the non-migrated state?

Reply:
I had this happen to me once when reloading the page. Waited for awhile, then noticed that the response to `getLoggedInUser` wasn't returning over the ws connection. I reloaded the page again, and it worked fine then.